src/FunctionApps/python/GenerateCSVs/elr.py

```diff
@@ -1,8 +1,39 @@
 from typing import List, Any
-from GenerateCSVs.patient import PATIENT_COLUMNS
+from GenerateCSVs.patient import PATIENT_COLUMNS, parse_patient_resource
 
-ELR_COLUMNS = PATIENT_COLUMNS + []
+
+ELR_COLUMNS = PATIENT_COLUMNS + ["loinc_code", "result", "effective_date"]
+
+
+def extract_covid_test(observation: dict) -> dict:
+    """
+    Given an observation FHIR profile, determine whether the profile
+    contains information related to a COVID test or COVID status.
+    If yes, collect the relevant lab code, test result, and test
+    date into a list.
+    """
+    print(observation)
+    code = observation["code"]["coding"][0]
+    if "loinc" in code["system"] and "valueCodeableConcept" in observation:
+        obs_date = observation["effectiveDateTime"]
+        result = observation["valueCodeableConcept"]["coding"][0]["display"]
+        return [code["code"], result, obs_date]
+    return []
```

Comment:
Man my types are all over the place today. You're absolutely correct about the `List[str]`. The length though is actually important to keep different, since that's how we check for a covid observation (I did some basic profiling and data exploration before I concluded that it was both necessary and sufficient for a lab observation to have a `loinc` code and the `valueCodingConcept`). See comment above about line 37.

Reply:
You could be right about effectiveDateTime though. Which would be really concerning, if that's the case 😕 . That's the only date like field that's consistently available across all observations. I think a deep dive on one or both our parts is probably warranted 😞 .

Reply:
Ahhh ooops I missed line 37 where we check the length.